include/nuttx/sensors/as5048a.h

```diff
@@ -0,0 +1,146 @@
+/****************************************************************************
+ * include/nuttx/sensors/as5048a.h
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+#ifndef __INCLUDE_NUTTX_SENSORS_AS5048A
+#define __INCLUDE_NUTTX_SENSORS_AS5048A
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+#include <nuttx/fs/ioctl.h>
+#include <nuttx/spi/spi.h>
+#include <nuttx/sensors/qencoder.h>
+
+#if defined(CONFIG_SPI) && defined(CONFIG_SENSORS_QENCODER) && defined(CONFIG_SENSORS_AS5048A)
```

Comment:
it's enough to check CONFIG_SENSORS_AS5048A

Reply:
thanks